src/components/toggle/toggle.js

```diff
@@ -0,0 +1,22 @@
+const toggles = document.querySelectorAll('[data-toggle]')
+
+function initToggle() {
+  if(!toggles.length) {
+    return
+  }
+
+  Array.prototype.forEach.call(toggles, toggle => {
+    const targetId = toggle.getAttribute('data-toggle-id')
+    const target = document.getElementById(targetId)
+
+    toggle.addEventListener('click', event => handleClick(event, target))
+  })
+
+  function handleClick(event, target) {
+    event.preventDefault()
+    history.replaceState('', document.title, window.location.pathname + window.location.search) // reset hash if for some reason it ended up in the url
+    target.classList.toggle('toggle--active')
```

Comment:
Aren't we losing a lot of functionality that was present in `site-navigation/site-navigation.js`?

Reply:
Only `focus-trap` I think. I just use a general approach for a toggle functionality that we can reuse everywhere. Do we want `focus-trap`?